daffodil-core/src/main/java/org/apache/daffodil/api/Daffodil.java

```diff
@@ -36,4 +49,192 @@ private Daffodil() {
+  }
+
+  /**
+   * {@link InfosetInputter} to read an infoset represented as JSON from a java.io.InputStream
+   * <p>
+   * Read in an infoset in the form of json text from a java.io.InputStream
+   *
+   * @param is the java.io.InputStream to read the json text from
+   * @return InfosetInputter
+   */
+  public static InfosetInputter newJsonInfosetInputter(InputStream is) {
+    return new org.apache.daffodil.runtime1.infoset.JsonInfosetInputter(is);
+  }
+
+  /**
+   * {@link InfosetInputter} to read an infoset represented as an org.jdom2.Document
+   *
+   * @param document the org.jdom2.Document infoset
+   * @return InfosetInputter
+   */
+  public static InfosetInputter newJDOMInfosetInputter(org.jdom2.Document document) {
+    return new org.apache.daffodil.runtime1.infoset.JDOMInfosetInputter(document);
+  }
+
+  /**
+   * {@link InfosetInputter} to read an infoset represented as an org.w3c.dom.Document
+   *
+   * @param document the org.w3c.dom.Document infoset. Note that w3c
+   *                 Documents are not guaranteed to be thread-safe, even if all
+   *                 users only read/traverse it. It is up to the user to ensure
+   *                 that the Document passed into the W3CDOMInfosetInputter is
+   *                 not read or written by other threads while the
+   *                 W3CDOMInfosetInputter has access to it.
+   * @return InfosetInputter
+   */
+  public static InfosetInputter newW3CDOMInfosetInputter(org.w3c.dom.Document document) {
+    return new org.apache.daffodil.runtime1.infoset.W3CDOMInfosetInputter(document);
+  }
+
+  /**
+   * Create an InputSourceDataInputStream from a java.io.InputStream
+   *
+   * @param is input stream to create from
+   * @return InputSourceDataInputStream from a java.io.InputStream
+   */
+  public static InputSourceDataInputStream newInputSourceDataInputStream(InputStream is) {
+    return org.apache.daffodil.io.InputSourceDataInputStream.apply(is);
+  }
+
+  /**
+   * Create an InputSourceDataInputStream from a java.nio.ByteBuffer
+   *
+   * @param bb byte buffer to create from
+   * @return InputSourceDataInputStream from a java.nio.ByteBuffer
+   */
+  public static InputSourceDataInputStream newInputSourceDataInputStream(ByteBuffer bb) {
+    return org.apache.daffodil.io.InputSourceDataInputStream.apply(bb);
+  }
+
+  /**
+   * Create an InputSourceDataInputStream from a byte array
+   *
+   * @param arr byte array to create from
+   * @return InputSourceDataInputStream from a byte array
+   */
+  public static InputSourceDataInputStream newInputSourceDataInputStream(byte[] arr) {
+    return org.apache.daffodil.io.InputSourceDataInputStream.apply(arr);
+  }
+
+  /**
+   * These are the events that a derived specific InfosetInputter
+   * creates.
+   * <p>
+   * The InfosetInputter base class figures out Daffodil InfosetEvents from
+   * the call-backs providing these derived-class events types.
+   */
+  public enum InfosetInputterEventType {
```

Comment:
It feels like this enum still wants to live in the `o.a.d.api.infoset` package, since this is something only used by custom InfosetInputter implementations, which mostly use APIs defined in `o.a.d.api.infoset`.